src/AI/WallyAI.java

```diff
@@ -647,9 +649,12 @@ private double findAssaultKills(GameMap gameMap, Map<XYCoord, Unit> neededAttack
         {
           neededAttacks.put(xyc, unit);
           double thisDamage = CombatEngine.simulateBattleResults(unit, target, gameMap, xyc.xCoord, xyc.yCoord).defenderHPLoss;
-          thisDamage = findAssaultKills(gameMap, neededAttacks, target, thisDamage);
+
+          if (thisDamage > target.getPreciseHP())
+            continue; // OHKOs should be decided using different logic
 
           log(String.format("  Use %s to deal %sHP?", unit.toStringWithLocation(), thisDamage));
+          thisDamage = findAssaultKills(gameMap, neededAttacks, target, thisDamage);
```

Comment:
Won't the printed value of `thisDamage` be inaccurate immediately after it's printed, since you are reassigning the value here?

Reply:
Yes, but it makes sense to print it here since that's the damage *this* unit can do, as opposed to the total so far.